Update local relay and parachain tutorials (#1140)
substrate-developer-hub/substrate-docs: content/md/en/docs/tutorials/connect-other-chains/rococo-slot.md

---
title: Rococo testnet slot acquisition
description: Acquire a slot on Rococo - a public test network for parachains
keywords:
  - rococo
  - testnet
  - faucet
  - relay chain
---

<!-- TODO NAV.YAML -->
<!-- TODO migrate to HTG section once live! This isn't a really a "tutorial" worthy item. -->

<!-- TODO Stand alone Rococo page instead of the wiki when ready. -->

[Rococo](https://wiki.polkadot.network/docs/build-pdk#rococo-testnet) is Parity's public test network for parachains.
This will guide you through how you can onboard your parachain to it.

## Before you begin

As Rococo is a shared testnet, you will need to [prepare some keys](#setup-a-wallet), [acquire a Rococo `ParaID`](#acquire-a-paraid) and configure your parachain beyond what is needed for local testing.
You will also need to:

1. Complete the [trusted network tutorial](/tutorials/get-started/trusted-network/), to understand custom keys chain spec modifications we need.
1. [Setup a wallet](#setup-a-wallet) for `ROC` tokens and ensure you have your seed words backed up!

> 1. Complete the [parachain tutorial](/tutorials/connect-other-chains/parachain) and **test your parachain on your locally** before you attempt _anything on Rococo_!

Author: this is explicitly needed. WE MUST have people test locally as ita a huge PITA for everyone when they just "try" on rococo. they fail

This way you are able to peer with Rococo validator nodes, otherwise you will not be able to produce blocks!
The peering port is set with the `--port <collator node>-- --port <relay node>` CLI flags, be sure to do this for _both_ nodes separately.
It's very likely you want at least your collator's `--ws-port <ws port>` port open as well to allow for yourself (and others) to connect with it via the Polkadot Apps UI or API calls.

## Register as a parathread

Before you can become a parachain, you must register as a parathread on Rococo:

1. Using your reserved `ParaID` from above, you can now register as a **parathread**.

![register-parathread.png](/media/images/docs/tutorials/09-cumulus/register-parathread.png)

1. Once your extrinsic succeeds, you will see a `registrar.Registered` event emitted.

![parathread-register-success.png](/media/images/docs/tutorials/09-cumulus/parathread-register-success.png)

1. Go to the [Parachains -> Parathreads](https://polkadot.js.org/apps/#/parachains/parathreads) page and you will see that your parathread registration is **Onboarding**:

![parathread-onboarding.png](/media/images/docs/tutorials/09-cumulus/parathread-onboarding.png)

> After the extrinsic succeeds, it takes 2 sessions for the chain to fully onboard as a parathread.

Author: https://wiki.polkadot.network/docs/glossary#session

## Request your parachain slot

Once the parachain is active as a parathread, the related project team should [open a request](https://github.com/paritytech/subport/issues/new?assignees=&labels=Rococo&template=rococo.yaml) for either a **permanent** or a **temporary parachain slot** on Rococo.

- **Permanent slots** are parachain slots assigned to teams who currently have a parachain slot on Polkadot (following a successful slot lease auction) and thus have the needs for continuously testing their codebase for compatibility with the latest bleeding edge features in a real-world live environment (Rococo).

>   There is a limited number of permanent slots made available (see note below).

Reviewer: ```suggestion
  Only a limited number of permanent slots are available (see note below).
```

> - **Temporary slots** are parachain slots that are dynamically allocated in a continuous rollover manner.

Reviewer: ```suggestion
- **Temporary slots** are parachain slots that are dynamically allocated.
```

>   Concretely, at every start of a lease period, a certain number of parathreads (up to a maximum defined in the relay chain's configuration) will be automatically upgraded to parachains for a certain duration.

Reviewer: ```suggestion
  At the start of every lease period, a certain number of parathreads—up to a maximum defined in the relay chain's configuration—are automatically upgraded to parachains for a certain duration.
```

>   The parachains that were active during the ending lease period will be automatically downgraded to parathreads to free the slots for others to use in the subsequent one.

Reviewer: ```suggestion
  The parachains that were active during the ending lease period are automatically downgraded to parathreads to free the slots for others to use in the subsequent period.
```

  The temporary slots are meant to be used by teams who do not have a parachain slot yet on Polkadot, and are aiming for one in the near future.

> The goal of this more dynamic allocations aims at helping teams to test their runtimes more often, and in a more streamlined manner.

Reviewer: ```suggestion
The goal dynamic allocations is to help teams test their runtimes more often, and in a more streamlined manner.
```

> > **NOTE:**

Reviewer: ```suggestion
NOTE:
```

> >

Reviewer: ```suggestion
```

> > Rococo's runtime sets `AssignSlotOrigin = EnsureRoot<Self::AccountId>;` at this time, thus ONLY someone with sudo access can assign slots!

Reviewer: ```suggestion
The Rococo runtime requires `sudo` access to assign slots (`AssignSlotOrigin = EnsureRoot<Self::AccountId>;`).
```

> >

Reviewer: ```suggestion
```

> > The Rococo `sudo` key is presently controlled by Parity Technologies at this time, so to have the operation required to get a slot, please go to the [Subport repo and open a `Rococo Slot Request`](https://github.com/paritytech/subport/issues/new?assignees=&labels=Rococo&template=rococo.yaml) once you complete the above and are ready to connect!

Reviewer: ```suggestion
The Rococo `sudo` key is currently controlled by Parity Technologies.
After you complete the steps in this tutorial and are ready to connect, go to the Subport repo and open a [`Rococo Slot Request`](https://github.com/paritytech/subport/issues/new?assignees=&labels=Rococo&template=rococo.yaml) to request a slot.
```

> > A Parity team member will initiate the slot for you ASAP.

Reviewer: ```suggestion
A Parity team member will respond with the slot for you to use.
```

> >

Reviewer: ```suggestion
```

> > Long-term, the plan is to make it a community-driven process via Rococo's governance framework.

Reviewer: ```suggestion
Eventually, this process is intended to be community-driven through the Rococo governance framework.
```

Using an account with the `AssignSlotOrigin` origin, follow these steps to assign a temporary slot:

> 1. In the [Polkadot-JS Apps for Rococo](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-rpc.polkadot.io#/extrinsics), go to `Developer -> Extrinsics`.

Reviewer: ```suggestion
1. Open [Polkadot-JS Apps for Rococo](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-rpc.polkadot.io#/extrinsics).
1. Click **Developer** > **Extrinsics**.
```

> 1. Select the account with enough `ROC` balance that you wish to use to submit a transaction with.

Reviewer: ```suggestion
1. Select the account you want to use to submit the transaction.
```

1. Select the `assignedSlots` pallet.
1. Choose the `assignTempParachainSlot` function.

> 1. Insert your reserved `ParaID`. Make sure this matches the one you reserved from the previous section!

Reviewer: ```suggestion
1. Insert your reserved `ParaID`. 
  Make sure you use the ParaID you previously reserved.
```

1. Select `Current` for the `LeasePeriodStart`. If the current slot is full, you will be assigned the next available slot.
1. Sign and submit the transaction.

Given a period lease duration of 1 day, the current settings for assigned parachain slots on Rococo are (at the time of writing):

- **Permanent slot least duration**: 1 year (365 days)
- **Temporary slot least duration**: 3 days
- **Maximum number of permanent slots**: up to 25 permanent slots
- **Maximum number of temporary slots**: up to 20 temporary slots
- **Maximum temporary slots allocated per leased period**: up to 5 temporary slots per 3-day temporary lease periods

These are subject to change, based on the needs of the community.

> 🎉**_Congratulations!_**🎉

Reviewer: ```suggestion
**_Congratulations!_**
```

> With your slot activated by the Parity team, you're now able to test your parachain on the Rococo test net!

Reviewer: ```suggestion
After your slot is activated by the Parity team, you can test your parachain on the Rococo test network.
```

> Please do note that when your temporary slot's lease ends, the parachain is automatically downgraded to a parathread. Registered and approved slots are cycled through automatically in a round-robin fashion, so you will expect to come back online as a parachain from time to time.

Reviewer: ```suggestion
Note that when your temporary slot's lease ends, the parachain is automatically downgraded to a parathread. Registered and approved slots are cycled through automatically in a round-robin fashion, so you can expect to come back online as a parachain from time to time.
```
